pyproject.toml

```diff
@@ -42,39 +42,34 @@ files = ["dowhy/__init__.py"]
 
 #
 # Dependency compatibility notes:
-# * autogluon requires scikit-learn <1.2.0
 # * numba (imported by econml) requires python <3.11
 #
 [tool.poetry.dependencies]
 python = ">=3.8,<3.11"
```

Comment:
we can change this to >=3.8,<3.12. the latest version of numba supports 3.11. 

can update line 45 too.